base.go

```diff
@@ -201,7 +209,7 @@ func InitDefaults(args *ReleaseType, chart, repo string, values interface{}) {
 	}
 
 	// Delete the HelmOptions input value -- it's not helpful and would cause a cycle.
-	delete(args.Values, FieldHelmOptionsInput)
+	delete(localMap, FieldHelmOptionsInput)
```

Comment:
Seems it would be simpler to just move this to the apply block above before returning?